From: Artem Kachitchkine <artem.kachitchkin@sun.com>
Date: Thu, 12 Oct 2006 17:03:08 +0000 (-0700)
Subject: do not let tools coredump if HAL is not running
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=c2053bf825cd9d81949d63ad83790a68310bb808

do not let tools coredump if HAL is not running

Make HAL tools actually fail with a useful message
in case HAL not running on a system.
---

--- a/tools/hal-device.c
+++ b/tools/hal-device.c
@@ -136,12 +136,16 @@ int main(int argc, char **argv)
 		return 2;
 	}
 
-	// fprintf(stderr, "connected to: %s\n", dbus_bus_get_unique_name(conn));
+	/* fprintf(stderr, "connected to: %s\n", dbus_bus_get_unique_name(conn)); */
 	if (!(hal_ctx = libhal_ctx_new())) return 3;
 	if (!libhal_ctx_set_dbus_connection(hal_ctx, conn)) return 4;
 	if (!libhal_ctx_init(hal_ctx, &error)) {
-		fprintf(stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 5;
 	}
 
--- a/tools/hal_find_by_capability.c
+++ b/tools/hal_find_by_capability.c
@@ -147,8 +147,12 @@ main (int argc, char *argv[])
 		return 1;
 	}
 	if (!libhal_ctx_init (hal_ctx, &error)) {
-		fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 1;
 	}
 
--- a/tools/hal_find_by_property.c
+++ b/tools/hal_find_by_property.c
@@ -154,8 +154,12 @@ main (int argc, char *argv[])
 		return 1;
 	}
 	if (!libhal_ctx_init (hal_ctx, &error)) {
-		fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 1;
 	}
 
--- a/tools/hal_get_property.c
+++ b/tools/hal_get_property.c
@@ -163,8 +163,12 @@ main (int argc, char *argv[])
 		return 1;
 	}
 	if (!libhal_ctx_init (hal_ctx, &error)) {
-		fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 1;
 	}
 
--- a/tools/hal_set_property.c
+++ b/tools/hal_set_property.c
@@ -230,8 +230,12 @@ main (int argc, char *argv[])
 		return 1;
 	}
 	if (!libhal_ctx_init (hal_ctx, &error)) {
-		fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 1;
 	}
 
--- a/tools/lshal.c
+++ b/tools/lshal.c
@@ -682,11 +682,12 @@ main (int argc, char *argv[])
 		return 1;
 	}
 	if (!libhal_ctx_init (hal_ctx, &error)) {
-		fprintf (stderr, "error: libhal_ctx_init: %s: %s\n",
-			 error.name, error.message);
-		fprintf (stderr, "Could not initialise connection to hald. \n "
-				 "Normally this mean the HAL daemon (hald) is not running or not ready.\n");
-		LIBHAL_FREE_DBUS_ERROR (&error);
+		if (dbus_error_is_set(&error)) {
+			fprintf (stderr, "error: libhal_ctx_init: %s: %s\n", error.name, error.message);
+			LIBHAL_FREE_DBUS_ERROR (&error);
+		}
+		fprintf (stderr, "Could not initialise connection to hald.\n"
+				 "Normally this means the HAL daemon (hald) is not running or not ready.\n");
 		return 1;
 	}
 

